### PR TITLE
Add summary of capabilities/user stories to docs index page

### PR DESCRIPTION
**Pull request by rosteen**
As it says in the title - I hope that this is something close to what you were imagining for the "user stories" on the docs landing page, to give people an idea of what they might use `specreduce` for. If I'm off the mark let me know.

### CODE REVIEW

**Review by ojustino — COMMENTED** [review]
Hi @rosteen, thanks for the pull request. I like the content and think it would be good to rearrange it. My suggestions:
- Leave the first sentence as is ("The specreduce package aims...").
- Combine the beginning of the second sentence (up to "...basic spectroscopic reduction") and your first sentence ("Specreduce is currently useful...") into something like "Specreduce is currently most useful for the following steps in basic spectroscopic reduction:"
- Leave your numbered list as is.
- Paste the rest of the original second sentence after the list – I think it's better to list what we can't do and where else to go only after what we've given details on what we can do.

**Comment by rosteen** [comment]
@ojustino I made some edits, does it sound better to you now?

File: docs/index.rst [hunk 1]
```diff
@@ -2,13 +2,28 @@
...
-examples will nevertheless be provided of its usage in conjunction with these
+basic spectroscopic reduction, currently encompassing the following three tasks:
+
+#. Determining the trace of a spectrum dispersed in a 2D image, either by setting a flat
+   trace or fitting a spline, polynomial, or other model to the positions of the dispersed
+   spectrum.
+#. Generating a background based on a region on one or both sides of this trace, and making
+   available the background image, 1D spectrum of the background, and the
+   background-subtracted image.
+#. Performing either a Horne (aka "optimal") or boxcar extraction on either the original
```
Comment (ojustino) [inline docs/index.rst:17]:
```suggestion
#. Performing either a Horne (a.k.a. "optimal") or boxcar extraction on either the original
```

File: docs/index.rst [hunk 2]
```diff
@@ -2,13 +2,28 @@
...
 and infrared spectroscopy, on which applications such as pipeline processes for
 specific instruments can be built. The scope of its functionality is limited to
-basic spectroscopic reduction, with basic *image* processing steps (such as
-bias subtraction) instead covered by `ccdproc <https://ccdproc.readthedocs.io/en/latest/>`_ and other packages, data
-analysis by `specutils <https://specutils.readthedocs.io/en/latest/>`_ and visualization by `matplotlib <https://matplotlib.org/>`_. A few
-examples will nevertheless be provided of its usage in conjunction with these
+basic spectroscopic reduction, currently encompassing the following three tasks:
+
+#. Determining the trace of a spectrum dispersed in a 2D image, either by setting a flat
+   trace or fitting a spline, polynomial, or other model to the positions of the dispersed
```
Comment (ojustino) [inline docs/index.rst:12]: I would also add that users can set a custom trace so `ArrayTrace` isn't left out.

File: docs/index.rst [hunk 3]
```diff
@@ -2,13 +2,28 @@
...
+#. Performing either a Horne (aka "optimal") or boxcar extraction on either the original
+   or background-subtracted 2D spectrum, using the trace generated by the first task to
+   generate a 1D spectrum.
+
+Further details about these capabilities are detailed in the sections linked below.
+Beyond these tasks, basic *image* processing steps (such as bias subtraction) are covered by
+`ccdproc <https://ccdproc.readthedocs.io/en/latest/>`_
+and other packages, data analysis by `specutils <https://specutils.readthedocs.io/en/latest/>`_,
+and visualization by `matplotlib <https://matplotlib.org/>`_. A few
+examples will be provided of using ``specreduce`` in conjunction with these
```
Comment (ojustino) [inline docs/index.rst:26]:
```suggestion
examples will be provided that feature ``specreduce`` in conjunction with these
```

**Review by ojustino — COMMENTED** [review]
This looks comprehensive to me. I have some assorted suggestions and am ready to approve once you've made decisions on them.

**Comment by rosteen** [comment]
@ojustino I put in all your suggestions, thanks!

**Review by ojustino — APPROVED** [review]
Thanks again, @rosteen.

**Pull request merged**